### ISSUE TITLE
Report when a Win32 listbox is multi-selection

### ISSUE THREAD

**Issue by mdlawler** [nvaccess/nvda#18365]
### Is your feature request related to a problem? Please describe.
no
### Describe the solution you'd like
I'd like NVDA to report when a Win32 listbox is multi-selection.  This would be helpful to know both that it is a listbox and that it is multi-selection.  Then the user would know they could multi-select contiguously with shift up and down arrow or noncontiguously with shift-f8, arrow to an item, and press space.  There are many  examples where this is still used including accevent.
### Additional context
This was a nice feature in Window-Eyes.

**Comment by Adriani90** [comment]
If this property is not provided by the accessibility API, NVDA cannot interpret by means of any heuristics whether the lisbox is multiselect or not.
I think this needs and external fix from Microsoft.
cc: @josephsl what do you think?

**Comment by mdlawler** [comment]
You can get this information from the style bits of the window.  I don't believe MSAA or UIA give this, but since it is a real Win32 control the style bits are your friend.

**Comment by josephsl** [comment]
Hi,

I don't think Microsoft can resolve this fully - Microsoft provides the API (including window style constants), and it is up to app developers to take advantage of features offered in the API. Remember that even though accessibility API's may have been defined by operating system vendors, it is up to app developers to implement needed accessibility improvements by taking advantage of the accessibility API's like UI Automation from the app itself (in this case, the app in question becomes the accessibility server).

For multi-select list boxes, this will require bitwise operations on window style constants and converting that into a state NVDA recognizes. This will involve a heuristic - not only checking accessibility API information, but making sure that it has all the needed window style bits set. This is possible because Win32 list boxes can set window style bits, and if NVDA recognizes Win32 list boxes properly, it is a matter of converting style bits to control states.

Thanks.

**Comment by josephsl** [comment]
Hi,

Reference: https://learn.microsoft.com/en-us/windows/win32/controls/list-box-styles

Thanks.

**Comment by Adriani90** [comment]
Also related: #7579.

**Comment by LeonarddeR** [comment]
Is there an easy to reproduce list view somewhere, so I can investigate next week?

**Comment by mdlawler** [comment]
All of the listboxes (note they are listboxes and not listviews) in the Microsoft accevent tool that is part of the Windows SDK demonstrate this issue.

**Comment by LeonarddeR** [comment]
Thank you.
